src/hooks/usePopularCities.js

```diff
@@ -0,0 +1,22 @@
+import { useQuery } from '@tanstack/react-query';
+import fetchPopularCities from '../services/PopularCitiesServise';
+import transformCityItem from '../data/transformCityItem';
+
+const popularCitiesQueryKey = 'popularCities';
+
+const usePopularCities = () => {
+  const { data, error, isLoading } = useQuery({
+    queryKey: [popularCitiesQueryKey],
+    queryFn: fetchPopularCities,
+  });
+
+  const transformedData = data?.map(transformCityItem);
```

Comment:
`useMemo` is useful here to avoid array recreating during state changes hidden in `useQuery`

Reply:
+ I usually add default value to avoid unexpected error outside the hook

```suggestion
  const transformedData = data?.map(transformCityItem) ?? [];
```